### DISCUSSION TITLE
External Track gantry ETH

### DISCUSSION THREAD

**Discussion by nikeftekhar** [visose/Robots#232] [Q&A] [answered]
Hi Vicente,

I have always used your pluggin for Robot communication. Very fan of your work!
So I have always used robots for static robot 3D printing, but know I would like to use the pluggin to do some robotic path simulation in the RFL of ETH. It is the first time that I use it with a external track system and I am obtaining unexpected results. My question is how do I control the movement of the track (gantry system). I am sending the extrusion toolpath targets as I have always done, but I do not know if now that there are 3 extra axis (3 gantry track axises) if I have to add extra commands or parameters or components in order to complete the robot command and not to obtain strange results, and if so, how its done. Thanks a lot in advance!
![Captura1](https://github.com/visose/Robots/assets/43377174/47e46fb7-245a-465b-9d8b-745f73bff6fc)

**Comment by visose** [comment, marked as answer]
Hi, this will depend on how the gantry is integrated. If it's setup as 3 external axes controlled by the same controller as the robot, then in your targets you must add the external axes values. To do this add the "External" input to the CreateTarget component and supply a list of 3 numbers, one per axis. You must manually specify the position of each axis per target. The simulation and the robot controller will not do "inverse kinematics" on all 9 axes, only on the 6 axis of the robot. If you want to move the base of the robot, you must explicitly supply the position of the 3 axes of the gantry on the targets you wish to move the robot.

**Comment by faithderenzis** [comment]
i sink...
